Dart/src/com/jetbrains/lang/dart/ide/runner/DartExecutionHelper.java

```diff
@@ -0,0 +1,115 @@
+  }
+
+  @SuppressWarnings("unused")
+  public static boolean hasIssues(@NotNull final Project project, @NotNull VirtualFile launchFile) {
+    return !getIssues(project, launchFile).isEmpty();
+  }
+
+  public static List<DartServerData.DartError> getIssues(@NotNull final Project project, @NotNull VirtualFile launchFile) {
+    return getIssues(project, launchFile, true);
+  }
+
+  public static List<DartServerData.DartError> getIssues(@NotNull final Project project,
+                                                         @NotNull VirtualFile launchFile,
+                                                         boolean onlyErrors) {
+    GlobalSearchScope scope = DartResolveScopeProvider.getDartScope(project, launchFile, true);
+    if (scope == null) {
+      return Collections.emptyList();
+    }
+
+    // Collect errors.
+    final DartAnalysisServerService analysisServerService = DartAnalysisServerService.getInstance(project);
+    List<DartServerData.DartError> errors = analysisServerService.getErrors(scope);
+    if (onlyErrors) {
+      errors = errors.stream().filter(DartServerData.DartError::isError).collect(Collectors.toList());
+    }
+
+    return errors;
+  }
+
+  @SuppressWarnings("unused")
+  public static void displayIssues(@NotNull final Project project, @NotNull VirtualFile launchFile) {
+    displayIssues(project, launchFile, "Analysis issues with launch", null);
+  }
+
+  @SuppressWarnings("SameParameterValue")
+  public static void displayIssues(@NotNull final Project project,
+                                   @NotNull VirtualFile launchFile,
+                                   @NotNull String launchTitle,
+                                   @Nullable Icon icon) {
+    clearIssueNotifications(project);
+
+    List<DartServerData.DartError> errors = getIssues(project, launchFile);
+
+    if (errors.isEmpty()) {
+      return;
+    }
+
+    final DartProblemsView problemsView = DartProblemsView.getInstance(project);
+
+    final String content;
+    if (errors.size() == 1) {
+      content = errors.get(0).getMessage() + " (<a href=\"issues\">show</a>)";
+    }
+    else {
+      content = errors.size() + " " + StringUtil.pluralize("issue", errors.size()) + " found. (<a href=\"issues\">show</a>)";
+      problemsView.showErrorNotification(project, launchTitle, content, icon);
+    }
+
+    // Show a notification on the dart analysis tool window.
+    problemsView.showErrorNotification(project, launchTitle, content, icon);
+
+    // Jump to and highlight the first error (order unspecified) in the editor.
+    DartServerData.DartError error = errors.get(0);
+    final VirtualFile errorFile = LocalFileSystem.getInstance().findFileByPath(error.getAnalysisErrorFileSD());
+    if (errorFile != null) {
+      final OpenFileDescriptor descriptor = new OpenFileDescriptor(project, errorFile, error.getOffset());
+      descriptor.setScrollType(ScrollType.MAKE_VISIBLE);
+      descriptor.navigate(true);
```

Comment:
Currently, there's no chance to disable jumping to the first error. Even turning the balloon off won't help. For Java, there's an option in Preferences. How about:
- do not jump to error in 2017.1.x;
- think about optional jumping for 2017.2 and give it a longer run during EAP cycle?
I'm afraid some users may find editor switching irritating :).